python/ffsim/protocols/trace_protocol.py

```diff
@@ -45,3 +51,87 @@ def trace(obj: Any, norb: int, nelec: tuple[int, int]) -> float:
         "The object did not have a _trace_ method that returned the trace "
         "or a _diag_ method that returned its diagonal entries."
     )
+
+
+def _trace_fermion_operator_single(
+    ferm: FermionOperator, norb: int, nelec: tuple[int, int]
+):
+    n_elec_alpha, n_elec_beta = nelec
+    op = list(ferm.keys())[0]
+    coeff = ferm[op]
+    alpha_indices = []
+    beta_indices = []
+    for _, spin, orb in op:
+        if not spin:
+            alpha_indices.append(orb)
+        else:
+            beta_indices.append(orb)
+    alpha_indices = list(set(alpha_indices))
+    beta_indices = list(set(beta_indices))
+    combined_indices = [(i, False) for i in alpha_indices] + [
+        (i, True) for i in beta_indices
+    ]
```

Comment:
Instead of creating separate alpha and beta sets and then combining them, just create a combined set from the beginning which stores `(orb, spin)` pairs. Since you need the size of the alpha and beta supports later, you can compute that at the same time if you want.